core/src/main/java/greencity/webcontroller/ManagementFactOfTheDayController.java

```diff
@@ -86,15 +87,10 @@ public String findAll(@RequestParam(required = false, name = "query") String que
     public GenericResponseDto saveFactOfTheDay(@Valid @RequestBody FactOfTheDayPostDTO factOfTheDayPostDTO,
                                                BindingResult bindingResult) {
         if (bindingResult.hasErrors()) {
-            GenericResponseDto genericResponseDto = new GenericResponseDto();
-            for (FieldError fieldError : bindingResult.getFieldErrors()) {
-                genericResponseDto.getErrors().add(
-                    new FieldErrorDto(fieldError.getField(), fieldError.getDefaultMessage()));
-            }
-            return genericResponseDto;
+            buildGenericResponseDto(bindingResult);
```

Comment:
Missing return statement.

Reply:
You can move if (bindingResult.hasErrors()) into buildGenericResponseDto as well.

Reply:
As buildGenericResponseDto is a generic functionality used in many places it might worth a unit test.

Reply:
Congratulations! Unit tests discovered the issue before it was merged.

Reply:
> You can move if (bindingResult.hasErrors()) into buildGenericResponseDto as well.

in that case how can i know should method call service or not?

Reply:
I would go as:
```
if (!bindingResult.hasErrors()) {
    placeService.save(placeAddDto, user.getEmail());
}
return GenericResponseDto.build(bindingResult); // note method rename
```

Reply:
> Missing return statement.

done

Reply:
> As buildGenericResponseDto is a generic functionality used in many places it might worth a unit test.

done

Reply:
> I would go as:
> 
> ```
> if (!bindingResult.hasErrors()) {
>     placeService.save(placeAddDto, user.getEmail());
> }
> return GenericResponseDto.build(bindingResult); // note method rename
> ```

done